src/lib.rs

```diff
@@ -108,38 +108,88 @@
 //! ```
 extern crate proc_macro;
 use std::path::{Path, PathBuf};
+
+#[cfg(feature = "shaderc")]
 use shaderc::{ShaderKind, SourceLanguage, OptimizationLevel, CompileOptions,
     TargetEnv, Compiler, EnvVersion};
+
+#[cfg(feature = "naga")]
+use naga::{valid::{ValidationFlags, Validator, Capabilities}, front::wgsl, back::spv, back::spv::WriterFlags};
+
+#[cfg(not(any(feature = "shaderc", feature = "naga")))]
+compile_error!("at least one compiler feature must be specified");
+
+enum InputSourceLanguage {
+    #[cfg(feature = "shaderc")]
+    ShaderC(SourceLanguage),
+
+    #[cfg(feature = "naga")]
+    WGSL
```

Comment:
Sounds like a mix of compilation backends and source languages. It's probably better to flatten `InputSourceLanguage::ShaderC(HLSL)` and `InputSourceLanguage::ShaderC(GLSL)` to `InputSourceLanguage::HLSL` and `InputSourceLanguage::GLSL`

Reply:
and done